packages/discord.js/src/client/websocket/WebSocketShard.js

```diff
@@ -559,6 +583,48 @@ class WebSocketShard extends EventEmitter {
     }, 20_000).unref();
   }
 
+  /**
+   * Sets the RESUMED dispatch timeout.
+   * @param {number} [time] If set to -1, it will clear the resumed timeout
+   * @private
+   */
+  setResumedDispatchTimeout(time) {
+    if (time === -1) {
+      if (this.resumedDispatchTimeout) {
+        this.debug('Clearing the RESUMED dispatch timeout.');
+        clearTimeout(this.resumedDispatchTimeout);
+        this.resumedDispatchTimeout = null;
+      }
+      return;
+    }
+    this.debug('Setting a RESUMED dispatch timeout for 30s.');
+    this.resumedDispatchTimeout = setTimeout(() => {
+      this.debug('Did not receive RESUMED in time. Destroying and connecting again.');
+      this.destroy({ reset: false, closeCode: 4009 });
+    }, 30_000).unref();
```

Comment:
I don't think we should call `.unref()` here, as this can potentially cause the process to exit early.

Also, why 30 seconds?

Reply:
> why 30 seconds?

As the RESUMED event is sent only after the missed events are replayed

Reply:
This is not a solution. Imagine if you resume and replay thousands of events... you'd end up in a case where the connection gets closed mid resume because of this interval!

Reply:
> This is not a solution! Imagine if you resume and replay thousands of events... you'd end up in a case where the connection gets closed mid resume because of this interval!

Fixing this would require either an interval to check the last packet timestamp, or resetting the timeout on every packet received until it resumed